### PR TITLE
ogdf: init at 2020.02

### PR DESCRIPTION
**Pull request by wavewave**
###### Motivation for this change
OGDF: Open Graph Drawing Framework C++ library is initially packaged.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/ogdf/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{ stdenv, fetchFromGitHub, cmake, doxygen }:
+
+stdenv.mkDerivation rec {
+  name = "ogdf-${version}";
```
Comment (mmahut) [inline pkgs/development/libraries/ogdf/default.nix]: Please, use ```pname```.

File: pkgs/development/libraries/ogdf/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  version = "2020.02";
+  src = fetchFromGitHub {
+    owner = "ogdf";
```
Comment (mmahut) [inline pkgs/development/libraries/ogdf/default.nix]:
```suggestion
    owner = pname;
```

File: pkgs/development/libraries/ogdf/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    repo = "ogdf";
```
Comment (mmahut) [inline pkgs/development/libraries/ogdf/default.nix]:
```suggestion
    repo = pname;
```

File: pkgs/development/libraries/ogdf/default.nix [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    rev = "be4452b6c36575d11a04d5bb1b362edeb8173eaa";
```
Comment (mmahut) [inline pkgs/development/libraries/ogdf/default.nix]: Please use a tag instead of revision.

File: pkgs/development/libraries/ogdf/default.nix [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+  hardeningDisable = [ "all" ];
+  nativeBuildInputs = [ cmake doxygen ];
+  cmakeFlags = [ "-DOGDF_WARNING_ERRORS=OFF"
+                 "-DCMAKE_CXX_FLAGS=-fPIC"
+               ];
+
+  meta = {
+    description = "Open Graph Drawing Framework/Open Graph algorithms and Data structure Framework";
+    homepage = http://www.ogdf.net;
```
Comment (mmahut) [inline pkgs/development/libraries/ogdf/default.nix]:
```suggestion
    homepage = "http://www.ogdf.net";
```

File: pkgs/development/libraries/ogdf/default.nix [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    sha256 = "0drrs8zh1097i5c60z9g658vs9k1iinkav8crlwk722ihfm1vxqd";
+  };
+
+  hardeningDisable = [ "all" ];
+  nativeBuildInputs = [ cmake doxygen ];
+  cmakeFlags = [ "-DOGDF_WARNING_ERRORS=OFF"
+                 "-DCMAKE_CXX_FLAGS=-fPIC"
+               ];
+
+  meta = {
```
Comment (mmahut) [inline pkgs/development/libraries/ogdf/default.nix]: Please use ```with stdenv.lib;```.

File: pkgs/development/libraries/ogdf/default.nix [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+  src = fetchFromGitHub {
+    owner = "ogdf";
+    repo = "ogdf";
+    rev = "be4452b6c36575d11a04d5bb1b362edeb8173eaa";
+    sha256 = "0drrs8zh1097i5c60z9g658vs9k1iinkav8crlwk722ihfm1vxqd";
+  };
+
+  hardeningDisable = [ "all" ];
+  nativeBuildInputs = [ cmake doxygen ];
+  cmakeFlags = [ "-DOGDF_WARNING_ERRORS=OFF"
```
Comment (mmahut) [inline pkgs/development/libraries/ogdf/default.nix]: What is the reason to use this flag?

Comment (wavewave) [reply pkgs/development/libraries/ogdf/default.nix]: thanks for catching this. no reason and i removed it.

**Comment by wavewave** [comment]
@mmahut I've changed as suggested. Thanks!

File: pkgs/development/libraries/ogdf/default.nix [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+  pname = "ogdf";
+  version = "2020.02";
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "catalpa-202002";
+    sha256 = "0drrs8zh1097i5c60z9g658vs9k1iinkav8crlwk722ihfm1vxqd";
+  };
+
+  hardeningDisable = [ "all" ];
```
Comment (veprbl) [inline pkgs/development/libraries/ogdf/default.nix]: Can you provide a more specific list of disabled hardening options? If you really sure that it must be "all", then can you, perhaps, add some comment explaining why you think this is the most appropriate?

Comment (wavewave) [reply pkgs/development/libraries/ogdf/default.nix]: Yes, I weakened it from `all` to `format`, and added a comment on where this became relevant. 
Thanks for checking this.

**Comment by veprbl** [comment]
@GrahamcOfBorg build ogdf

**Comment by veprbl** [comment]
@GrahamcOfBorg build ogdf

**Review by veprbl — APPROVED** [review]

**Comment by veprbl** [comment]
@wavewave Thank you!

**Pull request merged**